metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -357,12 +359,20 @@ def _make_bigquery_client(self) -> List[GCPLoggingClient]:
         # why we disable gRPC here.
         client_options = self.config.extra_client_options.copy()
         client_options["_use_grpc"] = False
-        project_id = self.config.project_id
+        project_id = self._choose_lineage_client_project_id()
         if project_id is not None:
             return [GCPLoggingClient(**client_options, project=project_id)]
         else:
             return [GCPLoggingClient(**client_options)]
 
+    def _choose_lineage_client_project_id(self) -> Optional[str]:
```

Comment:
Minor: I'd make this `_get` for consistency and clarity (since this method doesn't actually accept any inputs)
Maybe this should be a property so the value is only computed once instead of on each function call?